### ISSUE TITLE
Only filling in fields that I can see - bug or feature?

### ISSUE THREAD

**Issue by benshep** [bitwarden/browser#437]
On [this page](https://caboodle.nationalbooktokens.com/caboodle-firsts-the-wicked-cometh), I noticed that no fields were filled in (from an "identity" type record) when the form fields were out of view. If I scrolled the page so that they were in view, they were filled in as expected. If I can see part of the form, only the part that's in view is filled in. I would expect the form to be filled in even when it's not in the viewport - is this a bug or a feature?

**Comment by kspearrin** [comment]
Part of auto-filling an identity form requires some rather liberal detection rules which could cause erroneous filling on forms that appear out of view. For this reason the current auto-filling features give context and priority only to the forms that are visible on screen.

**Comment by benshep** [comment]
Fair enough, I thought it was probably that. Is it too difficult to decide if the reason a form is out of view is because it's hidden or just scrolled away?

**Comment by kspearrin** [comment]
Currently we cannot detect that. This can also be suggested as a security feature since a malicious website could try to gather more information about you by hiding forms off screen that would then be filled by bitwarden. The detection method we have no prevents that.

**Comment by benshep** [comment]
Yes, that's an excellent point. How about making it an option, i.e. a checkbox to say "Don't fill in non-visible fields"? And maybe for critical data (like credit cards), you are not able to uncheck the box (i.e. same behaviour as now).

**Comment by kspearrin** [comment]
I suppose we could make than an option somehow.

**Comment by gene1wood** [comment]
I've requested this [in this forum post](https://community.bitwarden.com/t/add-option-to-allow-filling-fields-that-arent-visible/4381)
